collector-sqs/src/main/java/zipkin/collector/sqs/SQSSpanProcessor.java

```diff
@@ -61,59 +61,74 @@
 
   @Override public void close() throws IOException {
     if (closed.getAndSet(true)) throw new IllegalStateException("SQS processor is already closed.");
+    synchronized (this) {
+      if (runner != null && !runner.isInterrupted()) runner.interrupt();
```

Comment:
Your suggestion is a much better approach, thanks for catching this!  I've refactored things to the while loop is tied to the `closed` property instead.  Also added a delayed client shutdown so it can try finishing any inflight requests before the client is cleaned up.